### DISCUSSION TITLE
🤔: ヘダーのログインボタン表示の実装について

### DISCUSSION THREAD

**Discussion by takei0w0** [ws-4020/mobile-app-crib-notes#150] [Q&A]
ToDoアプリのログイン画面で、ログインボタンの表示実装に違和感があったので質問です。

```jsx
  /* ～省略～ */
const createScreenOptions: (props: {
  route: RouteProp<RootStackParamList, keyof RootStackParamList>;
}) => StackNavigationOptions = ({route}) => {
  if (route.name === 'Login') {
    return {};
  } else {
    return {headerRight: () => <HeaderRight />};
  }
};
  /* ～省略～ */
```

画面名でif文場合分けするよりは、Screenのoptionで以下のように指定するのが自然じゃないかと思ったのですが
このような実装にした背景は何かありますか？

```jsx
    <nav.Screen
        name="Login"
        component={Login}
        options={{
          headerTitle: 'ログイン',
          headerRight: undefined,
        }}
    />
```

**Comment by creature-water-valley** [comment]
`headerTitle`に設定するように修正済み
[ヘッダータイトルのカスタマイズ
](https://ws-4020.github.io/mobile-app-crib-notes/react-native/learn/todo-app/screens/stack#%E3%83%98%E3%83%83%E3%83%80%E3%83%BC%E3%82%BF%E3%82%A4%E3%83%88%E3%83%AB%E3%81%AE%E3%82%AB%E3%82%B9%E3%82%BF%E3%83%9E%E3%82%A4%E3%82%BA)
